projects/igniteui-angular/src/lib/grid/column-moving.spec.ts

```diff
@@ -0,0 +1,1391 @@
+import { DebugElement } from '@angular/core';
+import { async, fakeAsync, TestBed, tick } from '@angular/core/testing';
+import { FormsModule } from '@angular/forms';
+import { By } from '@angular/platform-browser';
+import { BrowserAnimationsModule, NoopAnimationsModule } from '@angular/platform-browser/animations';
+import { IgxGridModule } from './index';
+import { IgxColumnGroupComponent } from './column.component';
+import { IgxInputDirective } from '../directives/input/input.directive';
+import { IgxToggleDirective } from '../directives/toggle/toggle.directive';
+import { SortingDirection } from '../data-operations/sorting-expression.interface';
+import {
+    MovableColumnsComponent,
+    MovableTemplatedColumnsComponent,
+    MovableColumnsLargeComponent,
+    MultiColumnHeadersComponent
+ } from '../test-utils/grid-samples.spec';
+import { UIInteractions, wait } from '../test-utils/ui-interactions.spec';
+
+const CELL_CSS_CLASS = '.igx-grid__td';
+const COLUMN_HEADER_CLASS = '.igx-grid__th';
+const COLUMN_GROUP_HEADER_CLASS = '.igx-grid__th--fw';
+
+describe('IgxGrid - Column Moving', () => {
+    beforeEach(async(() => {
+        TestBed.configureTestingModule({
+            declarations: [
+                MovableColumnsComponent,
+                MovableTemplatedColumnsComponent,
+                MovableColumnsLargeComponent,
+                MultiColumnHeadersComponent
+            ],
+            imports: [
+                FormsModule,
+                BrowserAnimationsModule,
+                NoopAnimationsModule,
+                IgxGridModule.forRoot()
+            ]
+        }).compileComponents();
+    }));
+
+    afterEach(() => {
+        UIInteractions.clearOverlay();
+    });
+
+    it('Should be able to reorder columns programmatically.', (() => {
+        const fixture = TestBed.createComponent(MovableColumnsComponent);
+        fixture.detectChanges();
+
```

Comment:
I would recommend extracting the component creation and getting the grid in a beforeEach, so the following code is not repeated in each test:
const fixture = TestBed.createComponent(MovableColumnsComponent);
fixture.detectChanges();
const grid = fixture.componentInstance.grid;

As there are a few components used (not only MovableColumnsComponent), there should be a separate describe for each test component.